WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsFragment.kt

```diff
@@ -24,33 +28,78 @@ class ReaderInterestsFragment : Fragment(R.layout.reader_interests_fragment_layo
-        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderInterestsViewModel::class.java)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+            .get(ReaderInterestsViewModel::class.java)
         startObserving()
     }
 
     private fun startObserving() {
         viewModel.uiState.observe(viewLifecycleOwner, Observer { uiState ->
             updateInterests(uiState.interestsUiState)
+            updateDoneButton(uiState.doneButtonUiState)
+        })
+
+        viewModel.navigateToDiscover.observe(viewLifecycleOwner, Observer { event ->
+            event?.getContentIfNotHandled()?.let {
+                navigateToDiscover()
+            }
         })
 
         viewModel.start()
     }
 
+    private fun updateDoneButton(doneButtonUiState: DoneButtonUiState) {
+        with(done_button) {
+            isEnabled = doneButtonUiState.enabled
+            text = getString(doneButtonUiState.titleRes)
+        }
+        uiHelpers.updateVisibility(done_button, doneButtonUiState.visible)
+    }
+
     private fun updateInterests(interestsUiState: List<InterestUiState>) {
         interestsUiState.forEachIndexed { index, interestTagUiState ->
-            val chip = layoutInflater.inflate(R.layout.reader_interest_filter_chip, interests_chip_group, false) as Chip
+            var chip = interests_chip_group.findViewWithTag<Chip>(interestTagUiState.title)
+            if (chip == null) {
+                chip = layoutInflater.inflate(
+                    R.layout.reader_interest_filter_chip,
+                    interests_chip_group,
+                    false
+                ) as Chip
+                with(chip) {
+                    layoutDirection = View.LAYOUT_DIRECTION_LOCALE
+                    tag = interestTagUiState.title
+                    setOnCheckedChangeListener { compoundButton, isChecked ->
+                        if (compoundButton.isPressed) {
+                            viewModel.onInterestAtIndexToggled(index, isChecked)
+                        }
+                    }
+                    interests_chip_group.addView(chip)
+                }
+            }
             with(chip) {
-                layoutDirection = View.LAYOUT_DIRECTION_LOCALE
                 text = interestTagUiState.title
-                setOnCheckedChangeListener { _, _ -> // TODO: Pass index to VM
-                }
+                isChecked = interestTagUiState.isChecked
             }
-
-            interests_chip_group.addView(chip)
         }
     }
+
+    private fun navigateToDiscover() {
+        val fragmentTransaction = parentFragmentManager.beginTransaction()
+        fragmentTransaction.setCustomAnimations(
+            R.anim.activity_slide_in_from_right, R.anim.activity_slide_out_to_left,
+            R.anim.activity_slide_in_from_left, R.anim.activity_slide_out_to_right
+        )
+        fragmentTransaction.replace(R.id.fragment_container, ReaderDiscoverFragment(), tag).commit()
+    }
```

Comment:
Not sure which transition animation we should use here. I've temporarily used slide animation.